hypershift-operator/controllers/nodepool/powervs.go

```diff
@@ -0,0 +1,123 @@
+package nodepool
+
+import (
+	"fmt"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	v1 "k8s.io/api/core/v1"
+	capipowervs "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultCOSRegion = "us-south"
+)
+
+// getImageRegion returns the nearest region os IBM COS bucket for the RHCOS images
+func getImageRegion(region string) string {
+	switch region {
+	case "dal", "us-south":
+		return "us-south"
+	case "eu-de":
+		return "eu-de"
+	case "lon":
+		return "eu-gb"
+	case "osa":
+		return "jp-osa"
+	case "syd":
+		return "au-syd"
+	case "sao":
+		return "br-sao"
+	case "tor":
+		return "ca-tor"
+	case "tok":
+		return "jp-tok"
+	case "us-east":
+		return "us-east"
+	default:
+		return defaultCOSRegion
+	}
+}
+
+func ibmPowerVSMachineTemplateSpec(nodePool *hyperv1.NodePool, ami string) *capipowervs.IBMPowerVSMachineTemplateSpec {
```

Comment:
what is an ami in IBM?

Reply:
In PowerVS it is called `BootImage` or just `Image`